core/creators/themedui.js

```diff
@@ -384,6 +388,63 @@ CKEDITOR.replaceClass = 'ckeditor';
 		return editor;
 	}
 
+	// Delay editor creation if given element is detached from DOM.
+	//
+	// Require `config.delayDetached` to be `true`.
+	//
+	// If `config.registerCallback` is declared as function, it will be invoked with single argument:
+	// function, that should be called to create editor.
+	//
+	// Otherwise: periodically (with `setInterval`) run check if element is attached to DOM and create editor automatically.
+	//
+	// ```js
+	//	CKEDITOR.replace( detachedEditorElement, {
+	//		delayDetached: true,
+	//		delayDetachedFrequency: 50,
+	//		registerCallback:RegisterCallback
+	//		}
+	//	} );
+	// ```
+	//
+	// @param {CKEDITOR.element} element The DOM element.
+	// @param {Object} [config] The specific configuration to apply to this
+	// editor instance. Configuration set here will override the global CKEditor settings.
+	// @param {String} [data] Initial value for the instance.
+	// @param {String} mode CKEditor mode.
+	function delayCreationOnDetachedElement( element, config, data, mode ) {
+		if ( !config || !config.delayDetached || !element.isDetached() ) {
+			return false;
+		}
+
+		if ( !config.registerCallback ) {
+			CKEDITOR.config.registerCallback( payload );
+			return true;
+		}
+
+		config.registerCallback( registerPayload );
+
+		return true;
+
+		function payload() {
+			var delay = parseFloat( config.delay );
+			delay = isNaN( delay ) ? CKEDITOR.config.delayDetachedFrequency : delay;
+
+			var intervalId = setInterval( function() {
+				if ( !element.isDetached() ) {
+					clearInterval( intervalId );
+
+					createInstance( element, config, data, mode );
+				}
+			}, delay );
+		}
+
+		function registerPayload() {
```

Comment:
Based on [https://github.com/ckeditor/ckeditor4/pull/4463#discussion_r586367355](https://github.com/ckeditor/ckeditor4/pull/4463#discussion_r586367355) I removed this function, and make its definition anonymous as `config.registerCallback` a parameter. Now it only wraps `createInstance` with proper params, so maybe there is no need to even name it?

Reply:
I'm not sure which lines are you referring to - my guess would be
```js
config.delayIfDetached_callback( function() {
			createInstance( element, config, data, mode );
		} );
```
but uses the config that we want to have 🤔  Could you clarify it please?

Reply:
Those are the lines I was writing about. Because earlier you asked me to rename `registerPayload` function.  
In [https://github.com/ckeditor/ckeditor4/pull/4463#discussion_r586367355](https://github.com/ckeditor/ckeditor4/pull/4463#discussion_r586367355) we decide to remove `isDetached` verify before further execution. So the only line which remains is the one with `createInstance`. And since we only need it to be wrapped in another function, and we don't use it anywhere else - it could be just placed as an argument.  
  
So I decline your 'rename' because now, there is no name needed ;) :P